File: LeetCode/201-300/283. 移动零.java
class Solution {
    public void moveZeroes(int[] nums) {
 int left = 0,right = 1;
        A: for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0 && right < nums.length) {
                while (right < nums.length && nums[right] == 0 ) {
                    if (++right == nums.length) break A;
                };
                nums[left++] = nums[right];
                nums[right++] = 0;
                if (right == nums.length) break;
            } else  {
                left++;
                right++;
            }
        }
    }
}

File: LeetCode/201-300/283. 移动零.go
func moveZeroes(nums []int)  {
    zeroCount := 0
    n := len(nums)
    res := make([]int, n)
    j:= 0
    for i:=0;i<n;i++ {
        if nums[i] == 0 {
            zeroCount++
        } else {
            res[j] = nums[i]
            j++
        }
    }
    for i:=0;i<n;i++ {
        nums[i] = res[i]
    }
    return
}   